extensions/amp-list/0.1/amp-list.js

```diff
@@ -923,19 +925,27 @@ export class AmpList extends AMP.BaseElement {
         }).then(() => {
           // Necessary since load-more elements are toggled in the above block
           this.attemptToFitLoadMore_(dev().assertElement(this.container_));
-        }).catch(() => {
-          this.mutateElement(() =>
-            this.getLoadMoreService_().setLoadMoreFailed())
-              .then(() => {
-                this.attemptToFitLoadMoreElement_(
-                    this.getLoadMoreService_().getLoadMoreFailedElement(),
-                    dev().assertElement(this.container_));
-                const loadMoreFailedClickable = this.getLoadMoreService_()
-                    .getLoadMoreFailedClickable();
-                this.unlistenLoadMore_ = listen(
-                    loadMoreFailedClickable,
-                    'click', () => this.loadMoreCallback_(/*opt_reload*/ true));
-              });
+        });
+  }
+
+  /**
+   * @private
+   */
+  handleLoadMoreFailed_() {
+    this.mutateElement(() => this.getLoadMoreService_().setLoadMoreFailed())
+        .then(() => {
+          this.attemptToFitLoadMoreElement_(
+              this.getLoadMoreService_().getLoadMoreFailedElement(),
+              dev().assertElement(this.container_));
+          const loadMoreFailedClickable = this.getLoadMoreService_()
```

Comment:
I'm not quite clear on why we need to listen/unlisten for clicks here. Is it to prevent multiple clicks to the load more? Adding/removing event handlers can be tricky, so I think it might be a bit more clear to always just listen for a click on the button, and then have the event handler do the right thing based on if there is a request in flight (or if we want to do what opt_reload does).